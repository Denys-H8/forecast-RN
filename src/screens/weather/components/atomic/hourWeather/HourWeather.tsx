import React from 'react';
import { View } from 'react-native';

import CustomText from 'src/components/text';
import Icon from 'react-native-vector-icons/Fontisto';
import { styles } from './hourWeather.styles';

type IProps = {
  hour: string;
  name: string;
  temperature: string;
};

const HourWeather: React.FC<IProps> = ({ hour, name, temperature }) => {
  return (
    <View style={styles.container}>
      <CustomText style={styles.hour}>{hour}</CustomText>
      <Icon name={name} size={28} color="#fff" />
      <View style={styles.containerDegree}>
        <CustomText style={styles.temperature}>{temperature}</CustomText>
        <CustomText style={styles.celsius}>0</CustomText>
      </View>
    </View>
  );
};

export default HourWeather;
