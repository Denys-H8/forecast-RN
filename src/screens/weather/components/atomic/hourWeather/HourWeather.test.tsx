import React from 'react';
import { render } from '@testing-library/react-native';
import HourWeather from './HourWeather';

const hour = '12:00';
const name = 'night-alt-cloudy';
const temperature = '18';

describe('HourWeather ', () => {
  test('should render HourWeather', () => {
    const { toJSON } = render(<HourWeather hour={hour} name={name} temperature={temperature} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
