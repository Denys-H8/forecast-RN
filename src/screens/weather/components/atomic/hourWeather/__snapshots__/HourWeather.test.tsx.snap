// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HourWeather  should render HourWeather 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "height": 90,
      "justifyContent": "space-between",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "fontFamily": "WorkSans-Regular",
        },
        Object {
          "color": "#8B8EA5",
          "fontSize": 12,
        },
      ]
    }
  >
    12:00
  </Text>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#fff",
          "fontSize": 28,
        },
        undefined,
        Object {
          "fontFamily": "Fontisto",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    î©‡
  </Text>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontSize": 20,
          },
        ]
      }
    >
      18
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontSize": 12,
            "lineHeight": 12,
          },
        ]
      }
    >
      0
    </Text>
  </View>
</View>
`;
